webpack.config.js

```diff
@@ -1,27 +1,59 @@
 const webpack = require('webpack');
-const {join} = require('path');
+
+class FallbackRequireWebpackPlugin {
+  apply(compiler) {
+	compiler.hooks.compilation.tap('FallbackRequireWebpackPlugin', (compilation) => {
+		const {renderRequire} = webpack.javascript.JavascriptModulesPlugin.getCompilationHooks(compilation);
+		renderRequire.tap(
+			'FallbackRequireWebpackPlugin',
+			(content, renderContext) => content.replace(
+				/\s*return module\.exports;\s*/,
+				'\nreturn (module.exports && module.exports.name === "webpackEmptyContext") ? require(moduleId) : module.exports;',
+			),
+		);
+	});
+  }
+}
 
 module.exports = {
+	mode: 'production',
 	target: 'node',
 	context: __dirname,
-	entry: './src/plop.js',
+	entry: {
+		main: {
+			import: './src/plop.js',
+			filename: './dist/plop.js',
+		},
+		bin: {
+			dependOn: 'main',
+			import: './src/bin/plop.js',
+			filename: './bin/plop.js',
+		},
+	},
 	output: {
-		path: join(__dirname, 'dist'),
-		filename: 'plop.js',
-		libraryTarget: 'commonjs2',
+		path: __dirname,
 	},
-	mode: 'production',
 	devtool: false,
-	optimization: {
-		minimize: true,
-	},
 	// Do not provide fake versions of process, __dirname, etc.
 	node: false,
 	resolve: {
 		extensions: ['.js'],
 	},
+	externalsPresets: {
+		node: true,
+	},
 	module: {
 		rules: [
+			// Replace require extension supports
+			{
+				test: /\.js$/,
+				loader: 'string-replace-loader',
+				options: {
+					search: '\\brequire\\.extensions\\b',
+					replace: '__non_webpack_require__.extensions',
+					flags: 'g',
+				},
+			},
```

Comment:
This replacer makes sure packages using the deprecated `require.extensions["ts"] = something` will not generate Webpack require code.

Reply:
Can we add this as a comment?

Also, I'm not sure I know of the depreciation or even what that syntax does - is there any further reading you could link me to catch up?

Reply:
Here are the node docs: https://nodejs.org/api/modules.html#modules_require_extensions

Webpack doesn't support this syntax and trying to bundle this results in wacky requires that error on runtime.